### ISSUE TITLE
Some passes are missed

### ISSUE THREAD

**Issue by astrojuanlu** [satellogic/orbit-predictor#99]
(Originally discovered by @fisadev)

```python
import datetime as dt

from orbit_predictor.sources import MemoryTLESource
from orbit_predictor.locations import Location

source = MemoryTLESource()
source.add_tle("TLE_ID", ('1 99999U 20003C   20266.16335194  .00000680  00000-0  27888-4 0  9992', '2 99999  97.3095 330.4275 0013971 101.1830 259.0981 15.27499258 38313'), dt.datetime(2020, 9, 22, 23, 5, 51, 159933))

loc = Location(
    name="loc", 
    latitude_deg=-15.137152171507697,
    longitude_deg=-0.4276612055384211,
    elevation_m=1.665102900005877e-05,
)

predictor = source.get_predictor("TLE_ID")

PASS_DATE = dt.datetime(2020, 9, 25, 9, 2, 6)
LIMIT_DATE = dt.datetime(2020, 9, 25, 10, 36, 0)

print(list(predictor.passes_over(
    loc, 
    when_utc=PASS_DATE,
    limit_date=LIMIT_DATE,
    aos_at_dg=0, max_elevation_gt=0,
)))
print(list(predictor.passes_over(
    loc, 
    when_utc=PASS_DATE - dt.timedelta(seconds=1),  # <-------------------
    limit_date=LIMIT_DATE,
    aos_at_dg=0, max_elevation_gt=0,
)))
# WTF
```

Result:

```
$ python prediction_fail.py 
[]
[<PredictedPass TLE_ID over loc on 2020-09-25 09:42:55.694555>]
```

Python 3.8 on Linux, both orbit-predictor 0.12.0 and `master`.

**Comment by astrojuanlu** [comment]
Just for reference, this is the answer given by Skyfield:

```python
from skyfield.api import EarthSatellite, Topos, load

ts = load.timescale()

line1 = '1 99999U 20003C   20266.16335194  .00000680  00000-0  27888-4 0  9992'
line2 = '2 99999  97.3095 330.4275 0013971 101.1830 259.0981 15.27499258 38313'
satellite = EarthSatellite(line1, line2, '99999U', ts)

loc = Topos(
    latitude_degrees=-15.137152171507697,
    longitude_degrees=-0.4276612055384211,
    elevation_m=1.665102900005877e-05
)

PASS_DATE = ts.utc(2020, 9, 25, 9, 2, 6)
LIMIT_DATE = ts.utc(2020, 9, 25, 10, 36, 0)

t, events = satellite.find_events(loc, PASS_DATE, LIMIT_DATE, altitude_degrees=0.0)
for ti, event in zip(t, events):
    name = ('rise above 0°', 'culminate', 'set below 0°')[event]
    print(ti.utc_strftime('%Y %b %d %H:%M:%S'), name)
```

```
$ python skyfield_pass.py 
2020 Sep 25 09:42:55 rise above 0°
2020 Sep 25 09:48:31 culminate
2020 Sep 25 09:54:15 set below 0°
```

**Comment by astrojuanlu** [comment]
The "instability" is here:

```
In [17]: lp = LocationPredictor(loc, predictor, start_date=None, limit_date=None, max_elevation_gt=0, aos_at_dg=0)

In [18]: lp._find_nearest_ascending(PASS_DATE)
Out[18]: datetime.datetime(2020, 9, 25, 10, 35, 22, 4424)

In [19]: lp._find_nearest_ascending(PASS_DATE - dt.timedelta(seconds=1))
Out[19]: datetime.datetime(2020, 9, 25, 9, 25, 24, 1106)
```

**Comment by astrojuanlu** [comment]
What's happening is:

```
In [34]: lp.is_ascending(PASS_DATE)                                                                                                                                                                                                           
Out[34]: False

In [35]: lp.is_ascending(PASS_DATE + dt.timedelta(seconds=6))                                                                                                                                                                                 
Out[35]: True
```

therefore, the `_step(0.99)` has this result:

```
In [36]: lp.is_ascending(PASS_DATE + lp._orbit_step(0.99))                                                                                                                                                                                    
Out[36]: True

In [37]: lp.is_ascending(PASS_DATE + lp._orbit_step(0.99) - dt.timedelta(seconds=1))                                                                                                                                                          
Out[37]: False
```

Looking at how `_sample_points` is defined:

https://github.com/satellogic/orbit-predictor/blob/e8edc02ac810c8d52cd3f102c9c08edc9cdf9c45/orbit_predictor/predictors/base.py#L447-L455

this logic is kind of backwards to me:

https://github.com/satellogic/orbit-predictor/blob/e8edc02ac810c8d52cd3f102c9c08edc9cdf9c45/orbit_predictor/predictors/base.py#L438-L441

In this case, the first `candidate` to evaluate is the furthest date, and therefore we "miss" one closer ascending time in the middle. Therefore, this is not finding the **nearest** ascending, as the function says.

In conclusion: this works by sheer luck.
